src/link.rs

```diff
@@ -20,45 +20,27 @@
 
 #[cfg(feature="runtime")]
 macro_rules! link {
-    (@IMPL: #[cfg($cfg:meta)] fn $name:ident($($pname:ident: $pty:ty), *) $(-> $ret:ty)*) => (
+    (@LOAD: #[cfg($cfg:meta)] fn $name:ident($($pname:ident: $pty:ty), *) $(-> $ret:ty)*) => (
         #[cfg($cfg)]
-        pub fn $name(library: &mut super::SharedLibrary, allow_failure: bool) -> Result<(), String> {
-            let symbol = unsafe { library.library.get(stringify!($name).as_bytes()) }.map_err(|_| {
-                format!("could not load `{}`", stringify!($name))
-            });
-            let symbol = if allow_failure {
-                symbol.ok().map(|s| *s)
-            } else {
-                Some(*try!(symbol))
-            };
-            library.functions.$name = symbol;
-            Ok(())
+        pub fn $name(library: &mut super::SharedLibrary) {
+            let symbol = unsafe { library.library.get(stringify!($name).as_bytes()) }.ok();
+            library.functions.$name = symbol.map(|s| *s);
         }
 
         #[cfg(not($cfg))]
-        pub fn $name(_: &mut super::SharedLibrary, _: bool) -> Result<(), String> {
-            Ok(())
-        }
+        pub fn $name(_: &mut super::SharedLibrary) {}
     );
 
-    (@IMPL: fn $name:ident($($pname:ident: $pty:ty), *) $(-> $ret:ty)*) => (
-        pub fn $name(library: &mut super::SharedLibrary, allow_failure: bool) -> Result<(), String> {
-            let symbol = unsafe { library.library.get(stringify!($name).as_bytes()) }.map_err(|_| {
-                format!("could not load `{}`", stringify!($name))
-            });
-            let symbol = if allow_failure {
-                symbol.ok().map(|s| *s)
-            } else {
-                Some(*try!(symbol))
-            };
-            library.functions.$name = symbol;
-            Ok(())
+    (@LOAD: fn $name:ident($($pname:ident: $pty:ty), *) $(-> $ret:ty)*) => (
+        pub fn $name(library: &mut super::SharedLibrary) {
+            let symbol = unsafe { library.library.get(stringify!($name).as_bytes()) }.ok();
+            library.functions.$name = symbol.map(|s| *s);
         }
     );
```

Comment:
Maybe this could be replaced with

```
    (@LOAD: fn $name:ident($($pname:ident: $pty:ty), *) $(-> $ret:ty)*) => (
        link!(@LOAD: #[cfg(feature="runtime")] fn $name($($pname: $pty), *) $(-> $ret)*);
    );
```

to reduce code duplication. Can you think of a more obvious "always true" cfg?